commit f1f0bdaa8d59511624af4f2d8040d1368d2a4dc6 (refs/changes/83/1083/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-12-07T22:30:06+00:00 (2 years, 10 months ago)
    
    joyent/node-lockfd#6 move flock change to 2.x stream, create v1.x branch for 1.x releases (v2.0.0)

diff --git a/CHANGES.md b/CHANGES.md
new file mode 100644
index 0000000..71f113a
--- /dev/null
+++ b/CHANGES.md
@@ -0,0 +1,35 @@
+# node-lockfd changelog
+
+## not yet released
+
+(nothing yet)
+
+## 2.0.0
+
+- joyent/node-lockfd#6 Re-branding the 1.1.0 release. The flock(3C)
+  support will be removed from the 1.x release stream (see the "v1.x"
+  branch).
+
+## 1.1.0
+
+*WARNING*: This release has been re-branded "2.0.0" because it broke
+compatibility by raising the mininum SmartOS platform requirement to one
+that includes flock(3C) -- OS-2868, platform versions after 20150218.
+
+- joyent/node-lockfd#5 Add support for flock(3C)
+
+## 1.0.1
+
+- Allow `require('lockfd')` import to work.
+
+## 1.0.0
+
+- Bump v8plus dep to 0.3.1.
+
+## 0.0.3
+
+- Improved error handling.
+
+## earlier
+
+See the commit log.
diff --git a/Makefile b/Makefile
index a066d30..c6a209e 100644
--- a/Makefile
+++ b/Makefile
@@ -67,6 +67,32 @@ test: $(TAPE)
 check:: $(ESLINT)
 	$(ESLINT) -c $(ESLINT_CONF) $(ESLINT_FILES)
 
+# Ensure CHANGES.md and package.json have the same version.
+.PHONY: versioncheck
+versioncheck:
+	@echo version is: $(shell cat package.json | json version)
+	[[ `cat package.json | json version` == `grep '^## ' CHANGES.md | head -2 | tail -1 | awk '{print $$2}'` ]]
+
+.PHONY: cutarelease
+cutarelease: versioncheck
+	[[ -z `git status --short` ]]  # If this fails, the working dir is dirty.
+	@which json 2>/dev/null 1>/dev/null && \
+	    ver=$(shell json -f package.json version) && \
+	    name=$(shell json -f package.json name) && \
+	    publishedVer=$(shell npm view -j $(shell json -f package.json name)@$(shell json -f package.json version) version 2>/dev/null) && \
+	    if [[ -n "$$publishedVer" ]]; then \
+		echo "error: $$name@$$ver is already published to npm"; \
+		exit 1; \
+	    fi && \
+	    echo "** Are you sure you want to tag and publish $$name@$$ver to npm?" && \
+	    echo "** Enter to continue, Ctrl+C to abort." && \
+	    read
+	ver=$(shell cat package.json | json version) && \
+	    date=$(shell date -u "+%Y-%m-%d") && \
+	    git tag -a "v$$ver" -m "version $$ver ($$date)" && \
+	    git push --tags origin && \
+	    npm publish
+
 #
 # Includes
 #
diff --git a/README.md b/README.md
index 3c7187e..8db701a 100644
--- a/README.md
+++ b/README.md
@@ -5,6 +5,11 @@
 
 # node-lockfd
 
+This repository is part of the Joyent Triton project. See the [contribution
+guidelines](https://github.com/joyent/triton/blob/master/CONTRIBUTING.md) --
+*Triton does not use GitHub PRs* -- and general documentation at the main
+[Triton project](https://github.com/joyent/triton) page.
+
 A trivial wrapper around [flock(3C)](https://illumos.org/man/3C/flock) and
 `fcntl(F_SETLKW)`. The provided interfaces presently allow a synchronous or
 asynchronous call to get a whole-file, exclusive, advisory write lock on a file,
@@ -101,6 +106,10 @@ mod_lockfd.flock(lockfileB, mod_lockfd.LOCK_EX, function (err) {
 });
 ```
 
+## Testing
+
+    make clean all test CTFCONVERT=/bin/true CTFMERGE=/bin/true
+
 ## License
 
 MIT.
diff --git a/package.json b/package.json
index 2d48522..4934b96 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "lockfd",
   "description": "thin wrapper around fcntl(F_SETLK) and flock()",
-  "version": "1.1.0",
+  "version": "2.0.0",
   "author": "Joshua M. Clulow <jmc@joyent.com>",
   "repository": "git://github.com/joyent/node-lockfd.git",
   "main": "./lib/index.js",
